Source/AwsCommonRuntimeKit/auth/imds/CRTIMDSClient.swift

```diff
@@ -2,7 +2,7 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import AwsCAuth
-// TODO: fix pointers and look for other errors
+// TODO: fix pointers and look for other errors, continue continuation if error during sync call, Use core class and Unmanaged
```

Comment:
This class has the previous buggy callback implementation of callbacks. I will fix it as part of future PR (Auth layer Refactor)